org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java

```diff
@@ -88,6 +91,37 @@ protected Resource loadResource(URI uri, ResourceSet localResourceSet, ResourceS
 						// (if we do try to load resources that are out of sync, we create a deadlock)
 						return null;
 					}
+				} else if (uri.isArchive()) {
+					String authority = uri.authority();
+					if (authority != null && authority.startsWith("platform:/resource/")) {
+						class MyArchiveURLConnection extends ArchiveURLConnection {
+							public MyArchiveURLConnection(String url) {
+								super(url);
+							}
+							public String getNestedURI() {
+								try {
+									return getNestedURL();
+								} catch (IOException exception) {
+									return ""; //$NON-NLS-1$
+								}
+							}
+						}
+						MyArchiveURLConnection archiveURLConnection = new MyArchiveURLConnection(uri.toString());
```

Comment:
Don't like the 'My' prefix. How about an anonymous class here?

Reply:
Same pattern is used a few lines below or above

Reply:
The anonymous class would not necessarily improve the readability since we need to access the nested URI, e.g. it would become something like
```
String nestedURIString = new ArchiveURLConnection(url) {
  public String getNestedURI() {..}
}.getNestedURI();
```
I'm not fond of this.

Reply:
But if we strip the exclamation mark from the authority, it should be safe to remove the url connection stuff completely.

Reply:
?

Reply:
I understood that the ArchiveURLConnection is only necessary to obtain the nestedURI. It would be sufficient to remove the trailing '!' from the authority and use URI.createURI on the remainder. With that at hand, you could start a new attempt to try isInSync. That would also make nested archives work.

Reply:
and i am not sure how to trim prefixes and suffixes so that uri.createuri wont explode

Reply:
and the api in emf that tells you how a archive uri starts and how no sucks even more that the one to test for platform resource uri

Reply:
:)
Agreed.